src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs

```diff
@@ -325,9 +325,17 @@ public override void Initialize([AllowNull] ViewWithData data)
                 var caption = localBranches.Count > 0 ?
                     "Checkout " + localBranches.First().DisplayName :
                     "Checkout to " + (await pullRequestsService.GetDefaultLocalBranchName(repository, Model.Number, Model.Title));
-                var disabled = await pullRequestsService.IsWorkingDirectoryClean(repository) ?
-                    null :
-                    "Cannot checkout as your working directory has uncommitted changes.";
+                var clean = await pullRequestsService.IsWorkingDirectoryClean(repository);
+                string disabled = null;
+
+                if (pullRequest.Head == null || !pullRequest.Head.RepositoryCloneUrl.IsValidUri)
+                {
+                    disabled = "The source repository is no longer available.";
```

Comment:
These should go into the resources file. Are we doing that on a later pass for all strings, or doing it as we go along? I remember we talked about this but not what we decided.